hoodie-common/src/main/java/com/uber/hoodie/common/table/log/block/metadata/HoodieMetadataType.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright (c) 2016 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.uber.hoodie.common.table.log.block.metadata;
+
+import com.google.common.collect.Maps;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Metadata abstraction for a HoodieLogBlock
+ *
+ */
+//This enum is serialized as the ordinal. Only add new enums at the end.
+//TODO : make this ordered and deterministic so it can evolve cleanly - NA
+public enum HoodieMetadataType {
+    INSTANT_TIME, TARGET_INSTANT_TIME;
+
+    public static byte [] convertToBytes(Map<HoodieMetadataType, String> metadata) throws IOException {
```

Comment:
still on the abstract class topic,  these classes would live inside `LogBlock` abstract class then.

Reply:
They do now, but still static because of the way LogBlocks are initalized.